code/game/machinery/poolcontroller.dm

```diff
@@ -0,0 +1,137 @@
+			emagged = 0 //Set the emagged var to false.
+			return
+		else
+			user << "\red Nothing happens." //If not emagged, don't do anything, and don't tell the user that it can be emagged.
+
+		return //Return, nothing else needs to be done.
+	else //If it's not a multitool, defer to /obj/machinery/proc/attackby
+		..()
+
+/obj/machinery/poolcontroller/attack_hand()
+	if(!Adjacent(usr))
+		usr << "You moved away."
+		return
+
+	if(emagged) //Emagging unlocks more (deadly) options.
+		var/temp = input("What temperature would you like to set the pool to?", "Pool Temperature") in list("Hot","Cold", "Warm", "Cool", "Normal","Cancel") //Allow user to choose which temperature they want.
+
+		switch(temp) //Used for setting the actual temperature var based on user input.
+			if("Hot")
+				miston()
+				temperature = "hot"
+				usr << "<span class='warning'>You flick the pool temperature to [temperature](WARNING).</span>" //Differ from standard message to make sure the user understands the temperature is harmful.
+				return
+			if("Cold")
+				temperature = "cold"
+				usr << "<span class='warning'>You flick the pool temperature to [temperature](WARNING).</span>" //Differ from standard message to make sure the user understands the temperature is harmful.
+				mistoff() //this won't get called otherwise
+				return
+
+			//Regular non-traitorous temperature choices still avalible.
+			if("Warm")
+				temperature = "warm"
+			if("Cool")
+				temperature = "cool"
+			if("Normal")
+				temperature = "normal"
+			if("Cancel")
+				return
+
+		mistoff()
+		usr << "<span class='warning'>You flick the pool temperature to [temperature].</span>" //Inform the mob of the temperature they just picked.
+		return
+
+	else
+		var/temp = input("What temperature would you like to set the pool to?", "Pool Temperature") in list("Warm","Cool","Normal","Cancel") //Allow user to choose which temperature they want.
+
+		switch(temp) //Used for setting the actual temperature var based on user input.
+			if("Warm")
+				temperature = "warm"
+			if("Cool")
+				temperature = "cool"
+			if("Normal")
+				temperature = "normal"
+			if("Cancel") //Cancel does nothing and leaves the temperature at it's previous state.
+				return
+
+		mistoff()
+		usr << "<span class='warning'>You flick the pool temperature to [temperature].</span>" //Display what the user picked.
+		return
+
+/obj/machinery/poolcontroller/process()
+	updateMobs() //Call the mob affecting proc
+
+/obj/machinery/poolcontroller/proc/updateMobs()
+	for(var/turf/simulated/floor/beach/water/W in linkedturfs) //Check for pool-turfs linked to the controller.
+		for(var/mob/M in W) //Check for mobs in the linked pool-turfs.
+			switch(temperature) //Apply different effects based on what the temperature is set to.
+				if("hot") //Burn the mob.
+					if(ishuman(M))
+						var/mob/living/carbon/human/H = M //If humanoid, directly apply fire damage for quicker effect.
+						H.adjustFireLoss(5)
```

Comment:
No need to cause fireloss directly; let damage be pushed onto the mob via  bodytemp.